### PR TITLE
RD-3100 Extended `HighlightText` component to accept all props supported by underlying `react-syntax-highlighter`

### PR DESCRIPTION
**Pull request by kubama**
## Description
In order to fix RD-3100 some customization to `HighlightText` is needed that is currently not supported.

## Screenshots / Videos
N/A

## Checklist

- [X] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires a documentation update.
- [X] I added proper labels to this PR.

## Tests
N/A

## Documentation
N/A

### CODE REVIEW

File: src/components/data/HighlightText/HighlightText.tsx [hunk 1]
```diff
@@ -25,33 +25,26 @@ SyntaxHighlighter.registerLanguage('yaml', yaml);
  * * JSON
  * * Python
  * * YAML
+ *
+ * All props supported by `react-syntax-highlighter` can be specified.
  */
-export default function HighlightText(props) {
-    const { children, language } = props;
-
+const HighlightText: FunctionComponent<SyntaxHighlighterProps> = ({
```
Comment (qooban) [inline src/components/data/HighlightText/HighlightText.tsx:31]: Having external `SyntaxHighlighterProps` how does the table with props in the documentation looks like?

I'm referring to: https://docs.cloudify.co/ui-components/2.7.1/?path=/docs/data-highlighttext--basic which has right now:
Screenshot from 2021-09-09 15-22-22

How will new storybook page for `HighlightText` look like?

File: src/components/data/HighlightText/HighlightText.tsx [hunk 2]
```diff
@@ -25,33 +25,26 @@ SyntaxHighlighter.registerLanguage('yaml', yaml);
...
+    children = '',
+    codeTagProps,
+    language = 'json',
```
Comment (qooban) [inline src/components/data/HighlightText/HighlightText.tsx:34]: As we register only predefined set of languages (see `SyntaxHighlighter.registerLanguage` calls), shouldn't we allow only that set of languages?

**Review by qooban — COMMENTED** [review]
Commented.

File: src/components/data/HighlightText/HighlightText.tsx (hunk 2 above)
Comment (kubama) [reply src/components/data/HighlightText/HighlightText.tsx:34]: We can always register additional languages, so I would say no.

File: src/components/data/HighlightText/HighlightText.tsx (hunk 1 above)
Comment (kubama) [reply src/components/data/HighlightText/HighlightText.tsx:31]: The new docs will not have the props listed. But the doc mentions which props are supported (I can add direct link).
This is something we already have in storybook, for example here: https://docs.cloudify.co/ui-components/2.7.1/?path=/docs/elements-checkmark--basic. The doc mentions that the component supports `Icon` props but does not list them. I don't see any issue with such approach.

Comment (qooban) [reply src/components/data/HighlightText/HighlightText.tsx:31]: As `SyntaxHighlighter` does not come from Semantic-UI-React, I think that having a link here is necessary. Thanks for adding it.

File: src/components/data/HighlightText/HighlightText.tsx (hunk 2 above)
Comment (qooban) [reply src/components/data/HighlightText/HighlightText.tsx:34]: Hmm... How widget developer would know about it?
I think we should either add a note about it to the description or allow just predefined set.

**Review by qooban — COMMENTED** [review]
Answered comments.

File: src/components/data/HighlightText/HighlightText.tsx (hunk 2 above)
Comment (qooban) [reply src/components/data/HighlightText/HighlightText.tsx:34]: Also, widget developer would need to import `react-syntax-highlighter` to call `registerLanguage` as we don't expose it through Stage API.

Comment (kubama) [reply src/components/data/HighlightText/HighlightText.tsx:41]: OK, I brought back limited set of values for this prop

**Review by qooban — APPROVED** [review]

**Pull request merged**